src/GenQPUtils.h

```diff
@@ -75,6 +75,43 @@ inline void fillQC(const std::vector<Task*>& tasks, int nrVars,
 	}
 }
 
+/**
+	* Reduce \f$ Q \f$ matrix and the \f$ c \f$ vector based on the dependencies list
+	*/
+inline void reduceQC(const Eigen::MatrixXd & QFull, const Eigen::VectorXd & CFull,
+										 Eigen::MatrixXd & Q, Eigen::VectorXd & C,
+										 const std::vector<int> & fullToReduced,
+										 const std::vector<int> & reducedToFull,
+										 const std::vector<std::tuple<int, int, double>> & dependencies)
+{
+	/* Start by moving the non-reduced variables to their new location */
+	for(size_t i = 0; i < reducedToFull.size(); ++i)
+	{
+		for(size_t j = 0; j < reducedToFull.size(); ++j)
+		{
+			Q(i,j) = QFull(reducedToFull[i], reducedToFull[j]);
+		}
+		C(i) = CFull(reducedToFull[i]);
+	}
```

Comment:
It could be done block-wise but that's not so trivial and it would probably be much harder to read. It could/should be reconsidered if/when Eigen introduces splicing operations.